### PR TITLE
Added some new features to the PR template

### PR DESCRIPTION
**Pull request by baggerspion**
This does not satisfy any issue.

**Description**:
Added some extra fields to the PR template to make it slightly richer. The CONTRIBUTING file will need a slight update to take this change into account.

**Tasks**:
N/A

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -1,13 +1,27 @@
...
+## Description
+- PR description message describing the motivation and context for the change
+  being proposed.
 
-**Tasks**:
+## Types of Changes
+_What types of changes does your code introduce? Put an `x` in all the boxes that apply:_
+- [ ] New feature (non-breaking change which adds functionality)
+- [ ] Bug fix (non-breaking change which fixes an issue)
+- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
```
Comment (princiya) [inline .github/PULL_REQUEST_TEMPLATE.md]: @therealpadams Do we need 'breaking' changes at all?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -1,13 +1,27 @@
...
+## Tasks
   - [x] Created Task 1
   - [x] Created Task 2
   - [ ] To-do Task 3
 
-**Screenshots/Videos**: (OPTIONAL)
+## Checklist
+_Go over all the following points, and put an `x` in all the boxes that apply:_
+- [ ] My change requires a change to the documentation & I have updated it accordingly.
+- [ ] My change requires updating the monitoring dashboard & I have updated it accordingly.
```
Comment (princiya) [inline .github/PULL_REQUEST_TEMPLATE.md]: It isn't clear what is a monitoring dashboard at this point. Do we need to add this now?

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (baggerspion) [reply .github/PULL_REQUEST_TEMPLATE.md]: Probably not; will remove this.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (baggerspion) [reply .github/PULL_REQUEST_TEMPLATE.md]: Nope, we don't.

**Review by princiya — APPROVED** [review]

**Pull request merged**